docs/source/tutorial_metrics.md

```diff
@@ -0,0 +1,372 @@
+`pip install nltk`.
+
+We can also measure ROUGE. Note that we need to `pip install py-rouge` for this
+functionality:
+
+```
+$ parlai eval_model -m fixed_response -t dailydialog --fixed-response "how may i help you ?" --metrics rouge
+14:47:24 | creating task(s): dailydialog
+14:47:31 | Finished evaluating tasks ['dailydialog'] using datatype valid
+    accuracy  exs    f1  rouge_1  rouge_2  rouge_L
+    .0001239 8069 .1163   .09887  .007285   .09525
+```
+
+### Agent-specific metrics
+
+Some agents include their own metrics that are computed for them. For example,
+generative models automatically compute `ppl`
+([perplexity](https://en.wikipedia.org/wiki/Perplexity)) and `token_acc`, both
+which measure the generative model's ability to predict indivdual tokens.  As
+an example, let's evaluate the [BlenderBot](https://parl.ai/projects/recipes/)
+90M model on DailyDialog:
+
+```
+$ parlai eval_model --task dailydialog -mf zoo:blender/blender_90M/model -bs 32
+...
+14:54:14 | Evaluating task dailydialog using datatype valid.
+14:54:14 | creating task(s): dailydialog
+...
+15:26:19 | Finished evaluating tasks ['dailydialog'] using datatype valid
+    accuracy  bleu-4  ctpb  ctps  exps  exs    f1  gpu_mem  loss      lr  ltpb  ltps   ppl  token_acc  total_train_updates   tpb   tps
+           0 .002097 14202 442.5 6.446 8069 .1345    .0384 2.979 7.5e-06  3242   101 19.67      .4133               339012 17445 543.5
+```
+
+Here we see a number of extra metrics, each of which we explain below:
+- `tpb`, `ctpb`, `ltpb`: stand for tokens per batch, context-tokens per batch,
+  and label-tokens per batch. These are useful for measuring how dense the
+  batches are, and are helpful when experimenting with [dynamic
+  batching](tutorial_fast).  tpb is always the sum of ctpb and lptb.
+- `tps`, `ctps`, `ltps`: are similar, but stand for "tokens per second". They
+  measure how fast we are training. Similarly, `exps` measures examples per
+  second.
+- `gpu_mem`: measures _roughly_ how much GPU memory your model is using, but it
+  is only approximate. This is useful for determining if you can possibly increase
+  the model size or the batch size.
+- `loss`: the loss metric
+- `ppl` and `token_acc`: the perplexity and per-token accuracy. these are generative
+  performance metrics.
+- `total_train_updates`: the number of SGD updates this model was trained for.
+  You will see this increase during training, but not during evaluation.
+
+## Adding custom metrics
+
+Of course, you may wish to add your own custom metrics: whether this is because
+you are developing a special model, special dataset, or otherwise want other
+information accessible to you. Metrics can be computed by either _the teacher_ OR
+_the model_. Within the model, they may be computed either _locally_ or _globally_.
+There are different reasons for why and where you would want to choose each
+location:
+
+- __Teacher metrics__: This is the best spot for computing metrics that depend
```

Comment:
I know you explain them below. But would you mind adding an example for each of these? It again helps with seeing them all in one place, and comparing them in one look.